presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -433,27 +453,49 @@ public synchronized void dropTable(HdfsContext context, String databaseName, Str
+            return getCommitHandle(databaseName + "." + tableName, operationStats);
+        });
     }
 
     public synchronized void renameTable(MetastoreContext metastoreContext, String databaseName, String tableName, String newDatabaseName, String newTableName)
     {
-        setExclusive((delegate, hdfsEnvironment) -> delegate.renameTable(metastoreContext, databaseName, tableName, newDatabaseName, newTableName));
+        setExclusive((delegate, hdfsEnvironment) -> {
+            MetastoreOperationStats operationStats = delegate.renameTable(metastoreContext, databaseName, tableName, newDatabaseName, newTableName);
+            return getCommitHandle(databaseName + "." + tableName, operationStats);
+        });
     }
 
     public synchronized void addColumn(MetastoreContext metastoreContext, String databaseName, String tableName, String columnName, HiveType columnType, String columnComment)
     {
-        setExclusive((delegate, hdfsEnvironment) -> delegate.addColumn(metastoreContext, databaseName, tableName, columnName, columnType, columnComment));
+        setExclusive((delegate, hdfsEnvironment) -> {
+            MetastoreOperationStats operationStats = delegate.addColumn(metastoreContext, databaseName, tableName, columnName, columnType, columnComment);
+            return getCommitHandle(databaseName + "." + tableName, operationStats);
+        });
     }
 
     public synchronized void renameColumn(MetastoreContext metastoreContext, String databaseName, String tableName, String oldColumnName, String newColumnName)
     {
-        setExclusive((delegate, hdfsEnvironment) -> delegate.renameColumn(metastoreContext, databaseName, tableName, oldColumnName, newColumnName));
+        setExclusive((delegate, hdfsEnvironment) -> {
+            MetastoreOperationStats operationStats = delegate.renameColumn(metastoreContext, databaseName, tableName, oldColumnName, newColumnName);
+            return getCommitHandle(databaseName + "." + tableName, operationStats);
+        });
     }
 
     public synchronized void dropColumn(MetastoreContext metastoreContext, String databaseName, String tableName, String columnName)
     {
-        setExclusive((delegate, hdfsEnvironment) -> delegate.dropColumn(metastoreContext, databaseName, tableName, columnName));
+        setExclusive((delegate, hdfsEnvironment) -> {
+            MetastoreOperationStats operationStats = delegate.dropColumn(metastoreContext, databaseName, tableName, columnName);
+            return getCommitHandle(databaseName + "." + tableName, operationStats);
+        });
+    }
+
+    private ConnectorCommitHandle getCommitHandle(String handleKey, MetastoreOperationStats operationStats)
+    {
+        HiveMetastoreCommitHandle commitHandle = new HiveMetastoreCommitHandle();
+        commitHandle.addLastDataCommitTimes(handleKey, operationStats.getLastDataCommitTimes());
```

Comment:
The right way is to maintain the map in this class and build the handle at the very end